pkg/services/accesscontrol/filter.go

```diff
@@ -0,0 +1,96 @@
+package accesscontrol
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+var sqlIDAcceptList = map[string]struct{}{}
+
+type SQLDialect interface {
+	DriverName() string
+}
+
+// Filter creates a where clause to restrict the view of a query based on a users permissions
+// Scopes for a certain action will be compared against prefix:id:sqlID where prefix is the scope prefix and sqlID
+// is the id to generate scope from e.g. user.id
+func Filter(ctx context.Context, dialect SQLDialect, sqlID, prefix, action string, user *models.SignedInUser) (string, []interface{}, error) {
+	if _, ok := sqlIDAcceptList[sqlID]; !ok {
+		return "", nil, errors.New("sqlID is not in the accept list")
+	}
+
+	var scopes []string
+	if user.Permissions != nil && user.Permissions[user.OrgId] != nil {
```

Comment:
Wouldn't `nil` permissions be an error here?

Reply:
Yeah that is true!